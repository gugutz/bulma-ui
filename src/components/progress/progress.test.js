import React from 'react'
import renderer from 'react-test-renderer'
import Progress from './progress'

describe('<Progress /> component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Progress value={60}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  const colors = ['primary', 'link', 'info', 'success', 'warning', 'danger']

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer
        .create(<Progress value={60} color={color}/>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer.create(<Progress value={60} size={size}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
