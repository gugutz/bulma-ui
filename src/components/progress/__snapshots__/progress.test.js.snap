// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> component should render correctly 1`] = `
<progress
  className="progress"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support color prop with danger 1`] = `
<progress
  className="progress is-danger"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support color prop with info 1`] = `
<progress
  className="progress is-info"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support color prop with link 1`] = `
<progress
  className="progress is-link"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support color prop with primary 1`] = `
<progress
  className="progress is-primary"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support color prop with success 1`] = `
<progress
  className="progress is-success"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support color prop with warning 1`] = `
<progress
  className="progress is-warning"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support size prop with large 1`] = `
<progress
  className="progress is-large"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support size prop with medium 1`] = `
<progress
  className="progress is-medium"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support size prop with normal 1`] = `
<progress
  className="progress"
  max="100"
  value={60}
>
  60%
</progress>
`;

exports[`<Progress /> component should support size prop with small 1`] = `
<progress
  className="progress is-small"
  max="100"
  value={60}
>
  60%
</progress>
`;
